### PR TITLE
Add new Paillier ek + proof to RefreshMessage.

### PR DESCRIPTION
**Pull request by tudorcebere**
Create new Paillier ek/dk pair + Add ek + proof of correctness to RefreshMessage.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -9,24 +9,27 @@ use curv::elliptic::curves::traits::{ECPoint, ECScalar};
...
     coefficients_committed_vec: VerifiableSS<P>,
     points_committed_vec: Vec<P>,
     points_encrypted_vec: Vec<BigInt>,
+    ek: EncryptionKey,
+    dk_correctness_proof: NICorrectKeyProof,
 }
 
 impl<P> RefreshMessage<P> {
-    pub fn distribute(old_key: &LocalKey<P>) -> Self
+    pub fn distribute(old_key: &LocalKey<P>) -> (Self, DecryptionKey)
```
Comment (omershlo) [inline src/lib.rs:32]: why not overwrite the existing `dk` in LocalKey?

File: src/lib.rs [hunk 2]
```diff
@@ -71,14 +74,21 @@ impl<P> RefreshMessage<P> {
...
-            fairness_proof_vec,
-            coefficients_committed_vec: vss_scheme,
-            points_committed_vec,
-            points_encrypted_vec,
-        }
+        let (ek, dk) = Paillier::keypair().keys();
+        let dk_correctness_proof = NICorrectKeyProof::proof(&dk, None);
+
+        (
+            RefreshMessage {
```
Comment (omershlo) [inline src/lib.rs:81]: previous `dk` must be zeroized and overwritten

File: src/lib.rs (hunk 1 above)
Comment (tudorcebere) [reply src/lib.rs:32]: Makes a ton of sense.

Comment (tudorcebere) [reply src/lib.rs:32]: The naming `old_key` made me think that a new one will be generated at some point. Maybe we would like to rename that only to local_key if we want to reuse it.

Comment (omershlo) [reply src/lib.rs:32]: good idea!

**Pull request merged**